### ISSUE TITLE
Persistence tests are failing

### ISSUE THREAD

**Issue by boc-tothefuture** [boc-tothefuture/openhab-jruby#182]
When I tried to merge the updates to persistence it started failing in the build (tried twice).  Not sure why it succeeded in the PR. 

Can you take a look @jimtng ?

**Comment by jimtng** [comment]
I'm looking into it.

**Comment by boc-tothefuture** [comment]
Now the status is showing as queued...  but i watched it fail twice.   Maybe something is off with the github actions.

**Comment by jimtng** [comment]
It passed on my local environment. It passed when I re-ran the tests on github. Is it a concurrency issue or timing issue related to persistence, because on github, multiple jobs were running for various PRs?

It does seem a bit touchy.

I'll submit a modified test to see if it would work better.

**Comment by boc-tothefuture** [comment]
shouldn't be any concurrency issue for jobs because the every job runs on its own VM.

**Comment by jimtng** [comment]
> shouldn't be any concurrency issue for jobs because the every job runs on its own VM.

Does each job have their own openhab installation? Does each openhab instance get set up from scratch and deleted on each run?

**Comment by boc-tothefuture** [comment]
https://github.com/boc-tothefuture/openhab-jruby/runs/1931022396?check_suite_focus=true
Shows a failure if you needed one.

**Comment by boc-tothefuture** [comment]
> > shouldn't be any concurrency issue for jobs because the every job runs on its own VM.
> 
> Does each job have their own openhab installation? Does each openhab instance get set up from scratch and deleted on each run?

Yes.  Essentially rake openhab:setup is run, then the openhab dir is zipped up and downloaded by each test VM, unzipped and tests are run against it.  

So its a fresh openhab for each feature file.

**Comment by jimtng** [comment]
Where can I find the Openhab logs in this step:

```
      - name: Upload OpenHAB Logs
        uses: actions/upload-artifact@v2
        if: failure()
        with:
          name: OpenHAB-logs-${{ github.workflow }}-${{ github.run_number }}-${{ matrix.feature }}
          path: tmp/
          retention-days: 2
```

**Comment by boc-tothefuture** [comment]
https://github.com/boc-tothefuture/openhab-jruby/actions/runs/579562773

I was just going to point to that.  :-) 
 
On any of the failed runs its on the bottom under artifacts.

**Comment by jimtng** [comment]
Succeeded twice, then failed after I've merged the change!

**Comment by boc-tothefuture** [comment]
Want me to take a look?

**Comment by boc-tothefuture** [comment]
At a quick glance, it could the timing of when the persistence is written to disk and when it comes back out.

Is there a check that isn't as time sensitive (the average since) seems very time sensitive.

**Comment by boc-tothefuture** [comment]
Can you access persistence values from the karaf console?

**Comment by jimtng** [comment]
I'm still working on it.

**Comment by jimtng** [comment]
Hopefully fixed by #186
